pydatastructs/miscellaneous_data_structures/disjoint_set.py

```diff
@@ -124,3 +124,24 @@ def find_size(self, key):
             raise KeyError("Invalid key, %s"%(key))
 
         return self.find_root(key).size
+
+    def groups(self):
+        """
+        Get the groups in DSU
+        """
+        result = dict()
+        for key, node in self.tree.items():
+            if node.parent == node:
+                parent = key
+            else:
+                parent = node.parent.key
```

Comment:
To be on a safe side, I'd recommend calling `find_root` so that we associate a given node with the correct root.